examples/linux/io_epoll_test.cpp

```diff
@@ -120,55 +111,43 @@ int main() {
 
   auto pipe_bench = [](auto& rPipeRef, auto& buffer, auto scheduler, int seconds,
                        auto& data, auto& reps, auto& offset) {
-    return defer(
-      [&, scheduler, seconds] {
+    return [&, scheduler, seconds] {
+      return [&] {
         return
-          transform_done(
-            stop_when(
-              // do reads
-              repeat_effect(
-                typed_via(
-                  defer(
-                    [&] {
-                      return transform(
-                        discard(
-                          async_read_some(
-                            rPipeRef,
-                            as_writable_bytes(span{buffer.data() + 0, 1}))),
-                        [&]{
-                          assert(data[(reps + offset)%sizeof(data)] == buffer[0]);
-                          ++reps;
-                        });
-                    }),
-                  scheduler)),
-              // stop reads after requested time
-              schedule_at(scheduler, now(scheduler) + std::chrono::seconds(seconds))),
-            // complete with void when requested time expires
-            []{return just();});
-      });
+          // do reads
+          async_read_some(rPipeRef, as_writable_bytes(span{buffer.data() + 0, 1}))
+          | discard
+          | transform([&] {
+              assert(data[(reps + offset) % sizeof(data)] == buffer[0]);
+              ++reps;
+            });
+      } | defer
```

Comment:
I do not have strong feelings about this. I think either has tradeoffs. 

I do not think that there is anything special about sender. I do think that | is especially expressive for any expression where the concept is maintained from LHS to RESULT.

If we took this as a policy, it would mean that defer, lazy, just should not be | enabled. This would expend to any algorithm that transitions from single to many (this applies to bulk and is resolved by the current proposal for having bulk_schedule() encompass a scope rather than introduce a scope that must be ended by bulk_join())

Reply:
Okay. more on this. 

The policy (| is for algorithms where the LHS concept is maintained in the RESULT) would also mean the sync_wait() should not support |

On the other hand, the approach taken in this PR would mean that when_all() and sequence() should add support for |. 

```cpp
  auto pipe_write = [](auto& wPipeRef, auto databuffer, auto scheduler, auto stopToken) {
    return
      // write the data to one end of the pipe
      std::make_tuple(
        [&]{
          printf("writes starting!\n");
        } | lazy,
        [&, databuffer] {
          return async_write_some(wPipeRef, databuffer) | discard;
        } | defer
          | typed_via(scheduler)
          | repeat_effect()
          | transform_done([]{return just();})
          | with_query_value(get_stop_token, stopToken),
        [&]{
          printf("writes stopped!\n");
        } | lazy)
          | sequence();
  };
```

Reply:
> The policy (`|` is for algorithms where the LHS concept is maintained in the RESULT) would also mean the `sync_wait()` should not support `|`

I had already been feeling that piping to `sync_wait` is dubious. I like the general guideline that a pipe should not change the concept of the things fed to it. Algorithms that accept and return senders get the pipe treatment, others don't.

> On the other hand, the approach taken in this PR would mean that when_all() and sequence() should add support for |.

This one doesn't make sense to me. Writing as `make_tuple(args...) | sequence()` seems worse in all respects to `sequence(args...)`.

Reply:
Perhaps the underlying guideline is more of "is this transforming an incoming monadic concept"?

Many of the 'head-of-pipeline' operations seem out-of-place using pipe-syntax.
e.g.
* `lambda | defer` feels odd, but `defer(lambda)` would seem better.
* `scheduler | schedule` feels odd but `schedule(scheduler)` seems better.

I don't necessarily have a problem with pipe operators changing the concept.
It would make sense to have a stream piping into a reduction algorithm and producing a sender, for example.

But maybe it doesn't make sense to have non-monadic things being used as LHS of pipe syntax?